components/notification.php

```diff
@@ -0,0 +1,350 @@
+<?php
+namespace Buddypress\CLI\Command;
+
+use WP_CLI;
+
+/**
+ * Manage BuddyPress Notifications.
+ *
+ * ## EXAMPLES
+ *
+ *     # Create notification item.
+ *     $ wp bp notification create
+ *     Success: Successfully created new notification. (ID #5464)
+ *
+ *     # Delete a notification item.
+ *     $ wp bp notification delete 520
+ *     Success: Notification deleted.
+ *
+ * @since 1.8.0
+ */
+class Notification extends BuddypressCommand {
+
+	/**
+	 * Object fields.
+	 *
+	 * @var array
+	 */
+	protected $obj_fields = array(
+		'id',
+		'user_id',
+		'item_id',
+		'secondary_item_id',
+		'component_name',
+		'component_action',
+		'date_notified',
+		'is_new',
+	);
+
+	/**
+	 * Create a notification item.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--component=<component>]
+	 * : The component for the notification item (groups, activity, etc). If
+	 * none is provided, a component will be randomly selected from the
+	 * active components.
+	 *
+	 * [--action=<action>]
+	 * : Name of the action to associate the notification. (comment_reply, update_reply, etc).
+	 *
+	 * [--user-id=<user>]
+	 * : ID of the user associated with the new notification.
+	 *
+	 * [--item-id=<item>]
+	 * : ID of the associated notification.
+	 *
+	 * [--secondary-item-id=<item>]
+	 * : ID of the secondary associated notification.
+	 *
+	 * [--date=<date>]
+	 * : GMT timestamp, in Y-m-d h:i:s format.
+	 * ---
+	 * default: Current time
+	 * ---
+	 *
+	 * [--silent]
+	 * : Whether to silent the notification creation.
+	 *
+	 * [--porcelain]
+	 * : Output only the new notification id.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp notification create --component=messages --action=update_reply --user-id=523
+	 *     Success: Successfully created new notification. (ID #5464)
+	 *
+	 *     $ wp bp notification add --component=groups --action=comment_reply --user-id=10
+	 *     Success: Successfully created new notification (ID #48949)
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$r = wp_parse_args( $assoc_args, array(
+			'component'         => '',
+			'action'            => '',
+			'user-id'           => 0,
+			'item-id'           => 0,
+			'secondary-item-id' => 0,
+			'date'              => bp_core_current_time(),
+		) );
+
+		// Fill in the component.
```

Comment:
As in the case of activity, I would suggest that we don't do this kind of magic. Better to have `component` as a required parameter, and then pass a random `component` in the `generate()` method.

Reply:
@boonebgorges Actually, we are not requiring on the activity command, see [here](https://github.com/buddypress/wp-cli-buddypress/blob/d5634e15c40710338634d481f4112282f09cb728/components/activity.php#L122).

Reply:
Ah, I misread. I think it's better in the long run to move randomness out of the `create` methods, but this doesn't have to be done now.

Reply:
Understood! I agree! There are some updates made here that needs to be replicated in other commands. I'll take this opportunity to update this as well. :)